src/Microsoft.Health.Fhir.Core/Features/Operations/Export/ExportJobTask.cs

```diff
@@ -354,6 +418,66 @@ private async Task<IAnonymizer> CreateAnonymizerAsync(CancellationToken cancella
                 queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, _exportJobRecord.ResourceType));
             }
 
+            var filteredResources = new HashSet<string>();
+            if (_exportJobRecord.Filters != null)
+            {
+                foreach (var filter in _exportJobRecord.Filters)
+                {
+                    filteredResources.Add(filter.ResourceType);
+                }
+            }
+
+            if (!progress.RunFilteredSearches)
+            {
+                var index = -1;
+                if (_exportJobRecord.Filters != null)
+                {
+                    foreach (var filter in _exportJobRecord.Filters)
+                    {
+                        index++;
+                        if (progress.CurrentFilter != null && filter != progress.CurrentFilter)
+                        {
+                            continue;
+                        }
+
+                        progress.SetFilter(filter);
+
+                        List<Tuple<string, string>> filterQueryParametersList = new List<Tuple<string, string>>(queryParametersList);
+                        filterQueryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type, filter.ResourceType));
+                        foreach (var param in filter.Parameters)
+                        {
+                            filterQueryParametersList.Add(param);
+                        }
+
+                        await SearchCompartmentWithFilter(exportJobConfiguration, progress, filter.ResourceType, filterQueryParametersList, batchIdPrefix + "-filter" + index, cancellationToken);
+
+                        progress.SetFilter(null);
+                    }
+                }
+
+                progress.MarkFiltersFinished();
+                await UpdateJobRecordAsync(cancellationToken);
+            }
+
+            foreach (var resource in filteredResources)
+            {
+                queryParametersList.Add(Tuple.Create(KnownQueryParameterNames.Type + ":not", resource));
```

Comment:
Assuming a sample request: `Patient/$export?_type=MedicationRequest&_typeFilter=MedicationRequest%3Fstatus%3Dactive`. Here we end up doing a search that has both _type=MedicationRequest and _type:not=MedicationRequest. This is an unnecessary search. Granted the export results won't be incorrect since we got the data that was requested (in the previous filtered search). But we should still avoid doing this search.